src/aof.c

```diff
@@ -1506,13 +1506,21 @@ int rewriteAppendOnlyFileRio(rio *aof) {
                 long long now = mstime();
                 if (now - updated_time >= 1000) {
                     sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, key_count, "AOF rewrite");
+                    key_count = 0;
                     updated_time = now;
                 }
             }
+
+            /* Delay before next key if required (for testing) */
+            if (server.rdb_key_save_delay)
+                debugDelay(server.rdb_key_save_delay);
         }
         dictReleaseIterator(di);
         di = NULL;
     }
+    /* Send one last child update if needed */
+    if (key_count)
+        sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, key_count, "AOF rewrite");
```

Comment:
in case of several small databases, we'll sent excessive amount of messages, and probe `/proc` many times (overhead).
but i suppose we can live with that.